### PR TITLE
If Yarn exists, don't show npm version warnings

### PR DESCRIPTION
**Pull request by peterpme**
Hey Expo,

Let me know if there's anything I can do to help push this along. I'd really love to be able to suppress the warning message for those that use `yarn` over `npm`.

Thanks!

### CODE REVIEW

File: src/project/Doctor.js [hunk 1]
```diff
@@ -41,8 +41,19 @@ function _isNpmVersionWithinRanges(npmVersion, ranges) {
 async function _checkNpmVersionAsync(projectRoot) {
   try {
     await Binaries.sourceBashLoginScriptsAsync();
+
+    try {
+      let yarnVersionResponse = await spawnAsync('yarn', ['--version']);
+      let yarnVersion = _.trim(yarnVersionResponse.stdout);
```
Comment (dikaiosune) [inline src/project/Doctor.js]: We're using modern-ish Node everywhere, so I think this should probably use [`String.prototype.trim()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim).

**Comment by anp** [comment]
Actually, thinking about it more, do we need to check the version field at all? I think we should instead call to `yarnpkg` (which won't be accidentally aliased to hadoop tools), check the exit code, and return `NO_ISSUES` if it is 0.

**Comment by peterpme** [comment]
Hey @dikaiosune,

Thanks for the review!

Are you saying we should run `yarnpkg` without `--version`? I chose `--version` strictly because I didn't want `yarn` to run. I'd be happy to update that. Let me know!

**Comment by ide** [comment]
I think what @dikaiosune is suggesting is to run `yarnpkg --version` but instead of checking stdout (the version), just checking the exit code.

**Comment by peterpme** [comment]
Gotcha! Something like this then?

``` es6
try {
    let yarnVersionResponse = await spawnAsync('yarnpkg', ['--version']);
    if (yarnVersionResponse.status === 0) {
      return NO_ISSUES
    }
  } catch (err) {}
```

**Comment by anp** [comment]
Yeah, I think that's about right

**Comment by davepack** [comment]
This has been merged.

**Comment by peterpme** [comment]
thank you!

**Pull request closed without merging**